OmniKit/MessageTransport/MessageBlocks/PodInfoFlashLogRecent.swift

```diff
@@ -10,59 +10,67 @@ import Foundation
 
 fileprivate let maxPumpEntriesReturned = 50
 
-// read (up to) the most recent 50 32-bit pump entries from flash log
-public struct PodInfoFlashLogRecent : PodInfo {
+// read (up to) the most recent 50 32-bit pump entries from pulse log
+public struct PodInfoPulseLogRecent : PodInfo {
     // CMD 1  2  3 4  5 6 7 8
     // DATA   0  1 2  3 4 5 6
     // 02 LL 50 IIII XXXXXXXX ...
 
-    public var podInfoType   : PodInfoResponseSubType = .flashLogRecent
+    public var podInfoType   : PodInfoResponseSubType = .pulseLogRecent
     public let data          : Data
-    public let indexLastEntry: UInt16 // how many 32-bit pump log entries total in Pod
-    public let hexWordLog    : Data   // TODO make a 32-bit pump log entry type
+    public let indexLastEntry: UInt16 // the last pulse entry
+    public var pulseLogEntry : [UInt32]
 
     public init(encodedData: Data) throws {
         if encodedData.count < 3 || ((encodedData.count - 3) & 0x3) != 0 {
             throw MessageBlockError.notEnoughData // first 3 bytes missing or non-integral # of log entries
         }
         let nLogBytesReturned = encodedData.count - 3
         let nLogEntriesReturned = nLogBytesReturned / 4
-        let lastPumpEntry = UInt16((encodedData[1] << 8) | encodedData[2])
+        let lastPumpEntry = (UInt16(encodedData[1]) << 8) | UInt16(encodedData[2])
         if lastPumpEntry < maxPumpEntriesReturned && nLogEntriesReturned < lastPumpEntry {
             throw MessageBlockError.notEnoughData // small count and we didn't recieve them all
         }
         self.data           = encodedData
         self.indexLastEntry = lastPumpEntry
-        self.hexWordLog     = encodedData.subdata(in: 3..<Int(encodedData.count))
+        self.pulseLogEntry  = Array(repeating: 0, count: nLogEntriesReturned)
+        var index = 0
+        while index < nLogEntriesReturned {
+            self.pulseLogEntry[index] = encodedData[(3+(index*4))...].toBigEndian(UInt32.self)
+            index += 1
+        }
     }
-    // TODO add code to nicely format the 32-bit pump log entries
 }
 
-// read (up to) the most previous 50 32-bit pump entries from flash log
-public struct PodInfoFlashLogPrevious : PodInfo {
+// read (up to) the most previous 50 32-bit pump entries from pulse log
+public struct PodInfoPulseLogPrevious : PodInfo {
```

Comment:
This seems to be largely a copy of the previous decoding. Can they be represented by the same decoding and the same structure?